app/controllers/controller.js

```diff
@@ -61,7 +62,36 @@ angular.module("umbraco").controller("Imulus.ArchetypeController", function ($sc
             });
         },
         update: function (ev, ui) {
+
+            // Variables.
+            var targetScope = ui.item.sortable.droptarget.scope();
+            var sourceScope = ui.item.scope();
+            var sameScope = sourceScope === targetScope;
+
+            // Special constraints for when moving between Archetypes.
+            if (sourceScope && !sameScope) {
+
+                // Variables.
+                var targetFieldsets = targetScope.model.config.fieldsets;
+                var index = ui.item.sortable.index;
+                var properties = sourceScope.model.value.fieldsets[index].properties;
+                var canRemove = sourceScope.canRemove();
+                var canAdd = targetScope.canAdd();
+                var inRange = canRemove && canAdd;
+                var isCompatible = propertiesMatchAnyFieldset(properties, targetFieldsets);
```

Comment:
@Nicholas-Westby Performance suggestion: No need to traverse all the properties of all fieldsets if `inRange` is `false` here.